flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusReporter.java

```diff
@@ -0,0 +1,256 @@
+		} else if (metric instanceof Counter) {
+			collector = createGauge((Counter) metric, validMetricName, metricIdentifier, dimensionKeys, dimensionValues);
+		} else if (metric instanceof Histogram) {
+			collector = createSummary((Histogram) metric, validMetricName, metricIdentifier, dimensionKeys, dimensionValues);
+		} else if (metric instanceof Meter) {
+			collector = createCounter((Meter) metric, validMetricName, metricIdentifier, dimensionKeys, dimensionValues);
+		} else {
+			log.error("Cannot add unknown metric type: {}. This indicates that the metric type is not supported by this reporter.",
+				metric.getClass().getName());
+			return;
+		}
+		collector.register();
+		collectorsByMetricName.put(metricName, collector);
+	}
+
+	@Override
+	public void notifyOfRemovedMetric(final Metric metric, final String metricName, final MetricGroup group) {
+		CollectorRegistry.defaultRegistry.unregister(collectorsByMetricName.get(metricName));
+		collectorsByMetricName.remove(metricName);
+	}
+
+	private Collector createGauge(final Gauge gauge, final String name, final String identifier, final List<String> labelNames, final List<String> labelValues) {
+		return io.prometheus.client.Gauge
+			.build()
+			.name(name)
+			.help(identifier)
+			.labelNames(toArray(labelNames, String.class))
+			.create()
+			.setChild(new io.prometheus.client.Gauge.Child() {
+				@Override
+				public double get() {
+					final Object value = gauge.getValue();
+					if (value instanceof Double) {
+						return (double) value;
+					}
+					if (value instanceof Number) {
+						return ((Number) value).doubleValue();
+					} else if (value instanceof Boolean) {
+						return ((Boolean) value) ? 1 : 0;
+					} else {
+						log.debug("Invalid type for Gauge {}: {}, only number types and booleans are supported by this reporter.",
+							gauge, value.getClass().getName());
+						return 0;
+					}
+				}
+			}, toArray(labelValues, String.class));
+	}
+
+	private static Collector createGauge(final Counter counter, final String name, final String identifier, final List<String> labelNames, final List<String> labelValues) {
+		return io.prometheus.client.Gauge
+			.build()
+			.name(name)
+			.help(identifier)
+			.labelNames(toArray(labelNames, String.class))
+			.create()
+			.setChild(new io.prometheus.client.Gauge.Child() {
+				@Override
+				public double get() {
+					return (double) counter.getCount();
+				}
+			}, toArray(labelValues, String.class));
+	}
+
+	private Collector createCounter(final Meter meter, final String name, final String identifier, final List<String> labelNames, final List<String> labelValues) {
+		return io.prometheus.client.Counter
+			.build()
+			.name(name + "_total")
+			.help(identifier)
+			.labelNames(toArray(labelNames, String.class))
+			.create()
+			.setChild(new io.prometheus.client.Counter.Child() {
+				@Override
+				public double get() {
+					return (double) meter.getCount();
```

Comment:
We should also export the rate of the meter as a prometheus gauge.

Personally i think we can drop this method completely; I'm not too big of a fan of exporting the count of a meter; if that would be of interest there should always be a separate counter for it. (and as far as Flink is concerned, there is)